### PR TITLE
Fix clippy::redundant_type_annotations lint, enable it

### LINKED ISSUE #161: Fix failures for lint clippy::redundant_type_annotations
# Clippy reports for lint clippy::redundant_type_annotations

Issues found:
<details>
<summary>
⚠️ redundant type annotation
 -- <a href="https://github.com/JustusFluegel/rust-ga/blob/8576fd7da643fd30f1e64b33d51b29a2afb7b063/packages/ec-linear/src/mutator/with_one_over_length.rs#L63C9-L63C76">packages/ec-linear/src/mutator/with_one_over_length.rs</a>
</summary>

```txt
warning: redundant type annotation
  --> packages/ec-linear/src/mutator/with_one_over_length.rs:63:9
   |
63 |         let parent_bits: Bitstring = Bitstring::random(num_bits, &mut rng);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_type_annotations
   = note: requested on the command line with `-W clippy::redundant-type-annotations`
```

</details>
<details>
<summary>
⚠️ redundant type annotation
 -- <a href="https://github.com/JustusFluegel/rust-ga/blob/8576fd7da643fd30f1e64b33d51b29a2afb7b063/packages/ec-linear/src/mutator/with_rate.rs#L76C9-L76C76">packages/ec-linear/src/mutator/with_rate.rs</a>
</summary>

```txt
warning: redundant type annotation
  --> packages/ec-linear/src/mutator/with_rate.rs:76:9
   |
76 |         let parent_bits: Bitstring = Bitstring::random(num_bits, &mut rng);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_type_annotations
```

</details>

### PR DESCRIPTION
**Pull request by JustusFluegel**
Closes #161

### CODE REVIEW

**Review by NicMcPhee — DISMISSED** [review]
I'm frankly amazed that these are the only two places that you needed to change to make this happy here!

**Comment by JustusFluegel** [comment]
> I'm frankly amazed that these are the only two places that you needed to change to make this happy here!

I think this lint only checks pretty basic cases :)

File: Cargo.toml [hunk 1]
```diff
@@ -84,8 +84,7 @@ allow_attributes = "deny"
 
 # allow the following lints for now until they are fixed
 cargo_common_metadata = "allow"
-multiple_crate_versions = "allow"
-use_debug = "allow"
+
```
Comment (NicMcPhee) [inline Cargo.toml:87]:
```suggestion

```

We don't want this blank line, do we?

**Review by NicMcPhee — CHANGES_REQUESTED** [review]

**Comment by JustusFluegel** [comment]
Same here

**Review by NicMcPhee — DISMISSED** [review]

**Pull request merged**